### PR TITLE
Export schemaType to file xrefmap.json

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#297519](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/297519)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6651)

### CODE REVIEW

File: src/docfx/build/xref/InternalXrefSpec.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ internal class InternalXrefSpec : IXrefSpec
     {
         public SourceInfo<string> Uid { get; }
 
+        public string? SchemaType { get; private set; }
```
Comment (yufeih) [inline src/docfx/build/xref/InternalXrefSpec.cs:15]: Add `string? SchemaType { get; }` to `IXrefSpec`?

File: src/docfx/lib/schema/JsonSchema.cs [hunk 2]
```diff
@@ -11,6 +11,11 @@ namespace Microsoft.Docs.Build
     [JsonConverter(typeof(JsonSchemaConverter))]
     internal class JsonSchema
     {
+        /// <summary>
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchema.cs:14]: nit: This is a custom extension, move it to line 193?

File: src/docfx/template/TemplateEngine.cs [hunk 3]
```diff
@@ -204,6 +204,7 @@ private RenderType GetTocRenderType()
             }
 
             var jsonSchema = JsonUtility.DeserializeData<JsonSchema>(schemaString, new FilePath(schemaFilePath));
+            jsonSchema.SchemaType = mime.ToLowerInvariant();
```
Comment (yufeih) [inline src/docfx/template/TemplateEngine.cs:207]: If `SchemaType` is a part of `JsonSchema` definition, that means people can override it in the schema definition. Is it intentional? If so, only assign the value if there is no existing `SchemaType`.

Comment (yufeih) [inline src/docfx/template/TemplateEngine.cs:207]: What is the purpose of to lower here? We typically preserve the case of mime.

Comment (xichan-ms) [reply src/docfx/template/TemplateEngine.cs:207]: OK.

File: src/docfx/build/xref/InternalXrefSpec.cs (hunk 1 above)
Comment (xichan-ms) [reply src/docfx/build/xref/InternalXrefSpec.cs:15]: OK.

File: src/docfx/lib/schema/JsonSchema.cs (hunk 2 above)
Comment (xichan-ms) [reply src/docfx/lib/schema/JsonSchema.cs:14]: OK.

File: docs/specs/schema.yml [hunk 4]
```diff
@@ -614,3 +614,24 @@ outputs:
...
+      "properties": {
+        "uid" : {"contentType": "uid"}
+      }
+    }
+outputs:
+  a.json:
+  .xrefmap.json: | 
+    {
+      "references":[
+        {"uid": "a", "schemaType": "testdata"}
```
Comment (OsmondJiang) [inline docs/specs/schema.yml:635]: >schemaType [](start = 22, length = 10)

how about call it "type"?

Comment (yufeih) [reply docs/specs/schema.yml:635]: That would likely collide with existing properties. Lots of properties we use is already called `type`

File: src/docfx/template/TemplateEngine.cs (hunk 3 above)
Comment (xichan-ms) [reply src/docfx/template/TemplateEngine.cs:207]: OK,  have removed it and added fallback logic.

File: docs/specs/schema.yml (hunk 4 above)
Comment (xichan-ms) [reply docs/specs/schema.yml:635]: OK, consider it in another PR.

File: src/docfx/lib/schema/JsonSchemaTransformer.cs [hunk 5]
```diff
@@ -175,7 +174,13 @@ public JToken TransformContent(ErrorBuilder errors, JsonSchema schema, FilePath
...
             var href = GetXrefHref(file, uid, uidCount, obj.Parent == null);
             var monikers = _monikerProvider.GetFileLevelMonikers(errors, file);
-            var xref = new InternalXrefSpec(uid, href, file, monikers, obj.Parent?.Path);
+            var xref = new InternalXrefSpec(
+                uid,
+                href,
+                file,
+                monikers,
+                obj.Parent?.Path,
+                schema == rootSchema ? schema.SchemaType is null ? _documentProvider.GetMime(file).Value : schema.SchemaType : null);
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchemaTransformer.cs:183]: Use `schema.SchemaType` when it is not root schema.

```suggestion
                schema.SchemaType ?? (schema == rootSchema ? _documentProvider.GetMime(file).Value : null);
```

Comment (xichan-ms) [reply src/docfx/lib/schema/JsonSchemaTransformer.cs:183]: OK.

**Review by yufeih — APPROVED** [review]
👍

**Pull request merged**